plutus-conformance/src/PlutusConformance/Common.hs

```diff
@@ -27,19 +28,34 @@ import UntypedPlutusCore.Evaluation.Machine.Cek (evaluateCekNoEmit)
 import UntypedPlutusCore.Parser qualified as UPLC
 import Witherable
 
-type UplcProg = UPLC.Program Name DefaultUni DefaultFun ()
-
-type UplcEvaluator = UplcProg -> Maybe UplcProg
+-- Common functions for all tests
 
--- | A UPLC evaluation test suite.
-data UplcEvaluationTest =
-    MkUplcEvaluationTest {
-       -- | The evaluator function to use for the test.
-       evaluator :: UplcEvaluator
-       -- | The test directory in which the test files are located.
-       , testDir :: FilePath
-    }
+-- | Walk a file tree, making test groups for directories with subdirectories,
+-- and test cases for directories without.
+discoverTests :: IsTest t =>
+    (FilePath -> t)
+    -> [String] -- ^ The list of tests that are to be labelled as `ExpectedFailure`.
```

Comment:
nit: I'd probably pass a `String -> Bool` predicate instead of a list of strings. It just adds a bit more generality and flexibility at very little cost.